hal/src/photon/concurrent_hal.cpp

```diff
@@ -127,3 +344,5 @@ void os_thread_scheduling(bool enabled, void* reserved)
     else
         taskEXIT_CRITICAL();
 }
+
```

Comment:
The logic is inverted in os_thread_scheduling if the meaning of os_thread_scheduling(false) is to disable the scheduler.